src/Analyzers/MSTest.Analyzers/UseClassCleanupBehaviorEndOfClassAnalyzer.cs

```diff
@@ -0,0 +1,89 @@
+
+    internal static readonly DiagnosticDescriptor UseClassCleanupBehaviorEndOfClassRule = DiagnosticDescriptorHelper.Create(
+        id: DiagnosticIds.UseClassCleanupBehaviorEndOfClassRuleId,
+        Title,
+        MessageFormat,
+        Description,
+        Category.Usage,
+        DiagnosticSeverity.Info,
+        isEnabledByDefault: true);
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; }
+        = ImmutableArray.Create(UseClassCleanupBehaviorEndOfClassRule);
+
+    public override void Initialize(AnalysisContext context)
+    {
+        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+        context.EnableConcurrentExecution();
+
+        context.RegisterCompilationStartAction(context =>
+        {
+            if (context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingClassCleanupAttribute, out INamedTypeSymbol? classCleanupAttributeSymbol)
+                 && context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingTestClassAttribute, out INamedTypeSymbol? testClassAttributeSymbol)
+                 && context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingClassCleanupBehavior, out INamedTypeSymbol? classCleanupBehaviorSymbol))
+            {
+                context.RegisterSymbolAction(
+                    context => AnalyzeSymbol(context, classCleanupAttributeSymbol, testClassAttributeSymbol, classCleanupBehaviorSymbol),
+                    SymbolKind.Method);
+            }
+        });
+    }
+
+    private static void AnalyzeSymbol(SymbolAnalysisContext context, INamedTypeSymbol classCleanupAttributeSymbol, INamedTypeSymbol testClassAttributeSymbol, INamedTypeSymbol classCleanupBehaviorSymbol)
+    {
+        var methodSymbol = (IMethodSymbol)context.Symbol;
+
+        if (!methodSymbol.ContainingType.GetAttributes().Any(x => x.AttributeClass.Inherits(testClassAttributeSymbol)))
+        {
+            return;
+        }
+
+        ImmutableArray<AttributeData> methodAttributes = methodSymbol.GetAttributes();
+        bool hasCleanupAttr = false;
+        bool hasCleanupEndOClassBehavior = false;
+        foreach (AttributeData methodAttribute in methodAttributes)
+        {
+            if (SymbolEqualityComparer.Default.Equals(methodAttribute.AttributeClass, classCleanupAttributeSymbol))
+            {
+                hasCleanupAttr = true;
+                foreach (TypedConstant arg in methodAttribute.ConstructorArguments)
+                {
+                    // one is the value for EndOFClass behavior in the CleanupBehavior enum.
+                    if (SymbolEqualityComparer.Default.Equals(arg.Type, classCleanupBehaviorSymbol)
+                        && arg.Value!.Equals(1))
```

Comment:
It's better to play safe and avoid `!`

```suggestion
                        && 1.Equals(arg.Value))
```